src/main/java/com/exactpro/epfast/decoder/message/instructions/PushContext.java

```diff
@@ -0,0 +1,30 @@
+/******************************************************************************
+ * Copyright 2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.exactpro.epfast.decoder.message.instructions;
+
+import com.exactpro.epfast.decoder.message.ExecutionContext;
+import com.exactpro.epfast.decoder.message.NormalInstruction;
+
+public class PushContext implements NormalInstruction {
```

Comment:
Combine `PushContext` with two `SetInstructionsXxx` to produced two `CallXxx` instructions.

I'd keep application message unchanged in call.